src/main/java/io/split/android/client/service/workmanager/SplitWorker.java

```diff
@@ -7,35 +7,89 @@
 import io.split.android.client.service.executor.SplitTask;
 import io.split.android.client.service.executor.SplitTaskExecutionInfo;
+import io.split.android.client.storage.db.SplitRoomDatabase;
+import io.split.android.client.utils.NetworkHelper;
+import io.split.android.engine.metrics.Metrics;
 
-import static com.google.common.base.Preconditions.checkNotNull;
+import static androidx.core.util.Preconditions.checkNotNull;
 
 public abstract class SplitWorker extends Worker {
 
+    private final SplitRoomDatabase mDatabase;
+    private final HttpClient mHttpClient;
+    private final NetworkHelper mNetworkHelper;
+    private final String mEndpoint;
+    private final Metrics mMetrics;
+
+    protected SplitTask mSplitTask;
 
-    SplitTask mSplitTask;
     public SplitWorker(@NonNull Context context,
-                       @NonNull WorkerParameters workerParams,
-                       @NonNull SplitTask task) {
+                       @NonNull WorkerParameters workerParams) {
 
         super(context, workerParams);
-        mSplitTask = checkNotNull(task);
+        Data inputData = workerParams.getInputData();
+        String databaseName = inputData.getString(ServiceConstants.WORKER_PARAM_DATABASE_NAME);
+        String apiKey = inputData.getString(ServiceConstants.WORKER_PARAM_API_KEY);
+        mEndpoint = inputData.getString(ServiceConstants.WORKER_PARAM_ENDPOINT);
+        String metricsEndpoint = inputData.getString(ServiceConstants.WORKER_PARAM_EVENTS_ENDPOINT);
+        mDatabase = SplitRoomDatabase.getDatabase(context, databaseName);
+
+        SplitHttpHeadersBuilder headersBuilder = new SplitHttpHeadersBuilder();
+        headersBuilder.setClientVersion(BuildConfig.VERSION_NAME);
+        headersBuilder.setApiToken(apiKey);
+        mHttpClient = new HttpClientImpl();
+        mHttpClient.addHeaders(headersBuilder.build());
+        mNetworkHelper = new NetworkHelper();
+
+        URI eventsRootTarget = URI.create(metricsEndpoint);
+        HttpMetrics httpMetrics = null;
+        try {
+            httpMetrics = HttpMetrics.create(mHttpClient, eventsRootTarget);
+        } catch (URISyntaxException e) {
+            e.printStackTrace();
+        }
+        mMetrics = new CachedMetrics(httpMetrics, TimeUnit.SECONDS.toMillis(1));
     }
 
     @NonNull
     @Override
     public Result doWork() {
-        SplitTaskExecutionInfo info = mSplitTask.execute();
-        Data data = new Data.Builder()
-                .putString(ServiceConstants.TASK_INFO_FIELD_STATUS, info.getStatus().toString())
-                .putString(ServiceConstants.TASK_INFO_FIELD_TYPE, info.getTaskType().toString())
-                .putInt(ServiceConstants.TASK_INFO_FIELD_RECORDS_NON_SENT, info.getNonSentRecords())
-                .putLong(ServiceConstants.TASK_INFO_FIELD_BYTES_NON_SET, info.getNonSentBytes())
-                .build();
-        return Result.success(data);
+        checkNotNull(mSplitTask);
```

Comment:
I don't have enough background, but is it okey to not use the result of `mSplitTask.execute()` in this update of `doWork`?

Reply:
Yes, that's ok. We're scheduling these tasks as periodic in work manager and we are not able to get the result from the SDK when active.